src/cargo/core/compiler/mod.rs

```diff
@@ -1420,7 +1421,9 @@ fn on_stderr_line_inner(
                 rendered: String,
                 message: String,
                 level: String,
+                children: Vec<Diagnostic>,
             }
+
```

Comment:
Should we hand-pick field to deserialize? I see other places doing so.

```suggestion
                children: Vec<Diagnostic>,
            }
            #[derive(serde::Deserialize)]
            struct Diagnostic {
                spans: Vec<Span>,
                // `children: Vec<Diagnostic>` if we need to check them recursively
            }
            #[derive(serde::Deserialize)]
            struct Span {
                suggestion_applicability: Option<Applicability>,
            }
```

BTW, I feel like every existing ad-hoc deserialization should have a comment telling which struct it refers to.

Reply:
> Should we hand-pick field to deserialize? I see other places doing so.

I'm not sure what you mean by this

Reply:
I mean deserializing only those fields we need instead of the whole Diagnostic struct

Reply:
We could do that but I was trying to use the available types from `rustfix`. I figured that since it had the ability to tell what things were it would be kept updated, instead of updating multiple places.

Reply:
If there is an update of types in `rustfix` crate, either reuse or not we still need to update to some extent. 

I was asking because other places deserialize fields-of-interest. Besides, there might be a performance hit if a package start emitting thousands of warnings with full struct fields, though I guess it is rare.

Anyway, I am good with reuse types from `rustfix`. Let's move on :)